apis/crd/v1alpha1/cloudentityselector_webhook.go

```diff
@@ -161,3 +169,76 @@ func (r *CloudEntitySelector) ValidateDelete() error {
 	// TODO(user): fill in your validation logic upon object deletion.
 	return nil
 }
+
+func (r *CloudEntitySelector) GetOwnerAccount() (*CloudProviderAccount, error) {
+	accountNameSpacedName := &types.NamespacedName{
+		Namespace: r.Namespace,
+		Name:      r.Spec.AccountName,
+	}
+	ownerAccount := &CloudProviderAccount{}
+	err := client.Get(context.TODO(), *accountNameSpacedName, ownerAccount)
+	if err != nil {
+		cloudentityselectorlog.Error(err, "failed to find owner account", "account", *accountNameSpacedName)
+		return nil, err
+	}
+	return ownerAccount, err
+}
+
+// validateMatchSections checks for unsupported match combinations and errors out
+func (r *CloudEntitySelector) validateMatchSections() error {
+	// MatchID and MatchName are not supported together in an EntityMatch, applicable for both vpcMatch, vmMatch section
+	for _, m := range r.Spec.VMSelector {
+		if m.VpcMatch != nil {
+			if len(strings.TrimSpace(m.VpcMatch.MatchID)) != 0 &&
+				len(strings.TrimSpace(m.VpcMatch.MatchName)) != 0 {
+				return fmt.Errorf("matchID and matchName are not supported together, " +
+					"configure either matchID or matchName in an EntityMatch")
+			}
+		}
+		for _, vmMatch := range m.VMMatch {
+			if len(strings.TrimSpace(vmMatch.MatchID)) != 0 &&
+				len(strings.TrimSpace(vmMatch.MatchName)) != 0 {
+				return fmt.Errorf("matchID and matchName are not supported together, " +
+					"configure either matchID or matchName in an EntityMatch")
+			}
+		}
+	}
+
+	ownerAccount, err := r.GetOwnerAccount()
+	if err != nil {
+		return err
+	}
+	cloudProviderType, err := ownerAccount.GetAccountProviderType()
+	if err != nil {
+		cloudentityselectorlog.Error(err, "Invalid cloud provider type")
+		return err
+	}
+
+	// In Azure, Vpc Name is not supported in vpcMatch
+	// In AWS, Vpc name(in vpcMatch section) with either vm id or vm name(in vmMatch section) is not supported
+	if cloudProviderType == AzureCloudProvider {
+		for _, m := range r.Spec.VMSelector {
+			if m.VpcMatch != nil {
+				if len(strings.TrimSpace(m.VpcMatch.MatchName)) != 0 {
+					return fmt.Errorf("vpc name match is not supported in vpcMatch, " +
+						"use vpc id instead of vpc name")
+				}
+			}
+		}
+	} else {
+		for _, m := range r.Spec.VMSelector {
+			if m.VpcMatch != nil {
+				if len(strings.TrimSpace(m.VpcMatch.MatchName)) != 0 {
+					for _, vmMatch := range m.VMMatch {
+						if len(strings.TrimSpace(vmMatch.MatchID)) != 0 ||
+							len(strings.TrimSpace(vmMatch.MatchName)) != 0 {
+							return fmt.Errorf("vpc name match with either vm id or vm name is not supported, " +
```

Comment:
Please keep the log message consistent.
Use `matchID` and `matchName` as you did in Line 194.